src/pages/Swap/index.tsx

```diff
@@ -143,13 +144,43 @@ function largerPercentValue(a?: Percent, b?: Percent) {
 
 const TRADE_STRING = 'SwapRouter'
 
-export default function Swap({ className }: { className?: string }) {
-  const navigate = useNavigate()
-  const { account, chainId } = useWeb3React()
+export default function SwapPage({ className }: { className?: string }) {
+  const { chainId: connectedChainId } = useWeb3React()
+  return (
+    <Trace page={InterfacePageName.SWAP_PAGE} shouldLogImpression>
+      <>
+        <PageWrapper>
+          <Swap className={className} chainId={connectedChainId} />
+          <NetworkAlert />
+        </PageWrapper>
+        <SwitchLocaleLink />
+      </>
+    </Trace>
+  )
+}
+
+/**
+ * The swap component displays the swap interface, manages state for the swap, and triggers onchain swaps.
+ *
+ * In most cases, chainId should refer to the connected chain, i.e. `useWeb3React().chainId`.
+ * However if this component is being used in a context that displays information from a different, unconnected
+ * chain (e.g. the TDP), then chainId should refer to the unconnected chain.
+ */
+export function Swap({
+  className,
+  prefilledState = {},
+  chainId,
+  onCurrencyChange,
+}: {
+  className?: string
+  prefilledState?: Partial<SwapState>
+  chainId: SupportedChainId | undefined
+  onCurrencyChange?: (selected: Pick<SwapState, Field.INPUT | Field.OUTPUT>) => void
+}) {
+  const { account, chainId: connectedChainId, connector } = useWeb3React()
   const loadedUrlParams = useDefaultsFromURLSearch()
```

Comment:
it still doesn't make sense to me that the URL parsing happens from within the shared component instead of on the page

imo this should be handled at the page level and the addresses should be passed in as part of `prefilledState`